ext/regex.go

```diff
@@ -330,3 +342,80 @@ func extractAll(target, regexStr ref.Val) ref.Val {
 	}
 	return types.NewStringList(types.DefaultTypeAdapter, result)
 }
+
+func estimateExtractCost() checker.FunctionEstimator {
+	return func(c checker.CostEstimator, target *checker.AstNode, args []checker.AstNode) *checker.CallEstimate {
+		if len(args) == 2 {
+			targetSize := estimateSize(c, args[0])
+			targetCost := targetSize.Add(checker.FixedSizeEstimate(1)).MultiplyByCostFactor(common.StringTraversalCostFactor)
+			regexCost := estimateSize(c, args[1]).Add(checker.FixedSizeEstimate(1)).MultiplyByCostFactor(common.RegexStringLengthCostFactor)
+			return &checker.CallEstimate{CostEstimate: checker.CostEstimate(targetCost).Multiply(regexCost), ResultSize: &targetSize}
+		}
+		return nil
+	}
+}
+
+func estimateExtractAllCost() checker.FunctionEstimator {
+	return func(c checker.CostEstimator, target *checker.AstNode, args []checker.AstNode) *checker.CallEstimate {
+		if len(args) == 2 {
+
+			targetSize := estimateSize(c, args[0])
+			targetCost := targetSize.Add(checker.FixedSizeEstimate(1)).MultiplyByCostFactor(common.StringTraversalCostFactor)
+			regexCost := estimateSize(c, args[1]).Add(checker.FixedSizeEstimate(1)).MultiplyByCostFactor(common.RegexStringLengthCostFactor)
+
+			resultSize := checker.SizeEstimate{Min: 0, Max: targetSize.Max}
```

Comment:
Nice, glad to see the extract all accounts for 0 as the min size and every char in the string as the max size. :)